lib/Jsrt/JsrtDebuggerObject.cpp

```diff
@@ -808,43 +822,51 @@ Js::JavascriptArray * JsrtDebugStackFrames::StackFrames(Js::ScriptContext * scri
 
     DWORD_PTR dispatchHaltFrameAddress = threadContext->GetDebugManager()->GetDispatchHaltFrameAddress();
     AssertMsg(dispatchHaltFrameAddress, "Didn't set the dispatchHaltFrameAddress at time of break?");
 
-    for (Js::ScriptContext *tempScriptContext = threadContext->GetScriptContextList();
-    tempScriptContext != nullptr && tempScriptContext->IsScriptContextInDebugMode();
-        tempScriptContext = tempScriptContext->next)
+    if (dispatchHaltFrameAddress != 0)
     {
-        Js::WeakDiagStack * framePointers = tempScriptContext->GetDebugContext()->GetProbeContainer()->GetFramePointers();
-        if (framePointers != nullptr)
+        typedef JsUtil::List<Js::DiagStackFrame*, ArenaAllocator> DiagStackFrameList;
+        DiagStackFrameList* stackList = Anew(this->jsrtDebugManager->GetDebugObjectArena(), DiagStackFrameList, this->jsrtDebugManager->GetDebugObjectArena(), 10);
+
+        // Walk all the script contexts and collect the frames which are below the address when break was reported.
+        for (Js::ScriptContext *tempScriptContext = threadContext->GetScriptContextList();
+        tempScriptContext != nullptr && tempScriptContext->IsScriptContextInDebugMode();
+            tempScriptContext = tempScriptContext->next)
         {
-            Js::DiagStack* stackFrames = framePointers->GetStrongReference();
-            if (stackFrames != nullptr)
+            Js::WeakDiagStack * framePointers = tempScriptContext->GetDebugContext()->GetProbeContainer()->GetFramePointers(dispatchHaltFrameAddress);
+            if (framePointers != nullptr)
             {
-                int count = stackFrames->Count();
-                for (int frameIndex = 0; frameIndex < count; ++frameIndex)
+                Js::DiagStack* stackFrames = framePointers->GetStrongReference();
+                if (stackFrames != nullptr)
                 {
-                    Js::DiagStackFrame* stackFrame = stackFrames->Peek(frameIndex);
-
-                    DWORD_PTR stackAddress = stackFrame->GetStackAddress();
-                    // Only give frames which were preset when break was triggered as we might have other user (non-library) scripts executed after break
-                    if (stackAddress >= dispatchHaltFrameAddress)
+                    int count = stackFrames->Count();
+                    for (int frameIndex = 0; frameIndex < count; ++frameIndex)
                     {
-                        Js::DynamicObject* stackTraceObject = this->GetStackFrame(stackFrame, frameCount);
-
-                        Js::Var marshaledObj = Js::CrossSite::MarshalVar(scriptContext, stackTraceObject);
-                        Js::JavascriptOperators::OP_SetElementI((Js::Var)stackTraceArray, Js::JavascriptNumber::ToVar(frameCount++, scriptContext), marshaledObj, scriptContext);
+                        Js::DiagStackFrame* stackFrame = stackFrames->Peek(frameIndex);
+                        stackList->Add(stackFrame);
```

Comment:
as you have remove the stackAddress >= dispatchHaltFrameAddress check, as I assume that this api will be called just after the break. Is this assumption will be valid all the time? Did you move this check somewhere? could we have assert just in case if it is not anywhere?